tests/test_compiler.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the compiler subpackage.
+TODO: Uncomment 'pytest.mark.external' to check these tests in GitHub actions with
+    the 'pennylane-catalyst' v0.3.2 release. These tests require the installation
+    of Catalyst from the main branch at the moment.
```

Comment:
I have no problem to wait for Catalyst release, but alternatively you can also pin Catalyst to master in the tests.

Reply:
@rmoyard What do you mean by pining Catalyst to master? We don't build wheels for the master branch.

Reply:
True, I forgot that!